api/api/test/test_authentication.py

```diff
@@ -72,44 +72,47 @@ def test_check_user(mock_raise_if_exc, mock_submit, mock_distribute_function, mo
-    result = authentication.generate_secret()
-    assert isinstance(result, str)
-    assert result == 'test_token'
-
-    calls = [call(authentication._secret_file_path, mode='x')]
-    mock_open.has_calls(calls)
-    mock_chown.assert_called_once_with(authentication._secret_file_path, authentication.wazuh_uid(),
-                                       authentication.wazuh_gid())
-    mock_chmod.assert_called_once_with(authentication._secret_file_path, 0o640)
+def test_generate_keypair(mock_open, mock_chown, mock_chmod, mock_change_keypair):
+    """Verify correct params when calling open method inside generate_keypair"""
+    result = authentication.generate_keypair()
+    assert isinstance(result[0], str)
+    assert isinstance(result[1], str)
+    assert result == ('-----BEGIN PRIVATE KEY-----',
+                      '-----BEGIN PUBLIC KEY-----')
+    
+    calls = [call(authentication._private_key_path, authentication.wazuh_uid(), authentication.wazuh_gid()),
+             call(authentication._public_key_path, authentication.wazuh_uid(), authentication.wazuh_gid())]
+    mock_chown.assert_has_calls(calls)
+    calls = [call(authentication._private_key_path, 0o640),
+             call(authentication._public_key_path, 0o640)]
+    mock_chmod.assert_has_calls(calls)
 
     with patch('os.path.exists', return_value=True):
-        authentication.generate_secret()
-
-        calls.append(call(authentication._secret_file_path, mode='r'))
+        calls = [call(authentication._private_key_path, mode='rb'), call(authentication._public_key_path, mode='rb')]
         mock_open.has_calls(calls)
 
 
-def test_generate_secret_ko():
+def test_generate_keypair_ko():
     """Verify expected exception is raised when IOError"""
     with patch('builtins.open'):
         with patch('os.chmod'):
-            with patch('api.authentication.chown', side_effect=PermissionError):
-                assert authentication.generate_secret()
+            with patch('os.chown', side_effect=PermissionError):
+                assert authentication.generate_keypair()
 
 
-@patch('api.authentication.token_urlsafe', return_value='test_token')
 @patch('builtins.open')
-def test_change_secret(mock_open, mock_token):
-    """Verify correct params when calling open method inside change_secret"""
-    authentication.change_secret()
-    mock_open.assert_called_once_with(authentication._secret_file_path, mode='w')
-    mock_open.return_value.__enter__().write.assert_called_once_with('test_token')
+def test_change_keypair(mock_open):
+    """Verify correct params when calling open method inside change_keypair"""
+    result = authentication.change_keypair()
+    assert isinstance(result[0], str)
+    assert isinstance(result[1], str)
+    calls = [call(authentication._private_key_path, mode='wb'), call(authentication._public_key_path, mode='wb')]
+    mock_open.has_calls(calls)
```

Comment:
Similar to the comment above, calls are not the right ones. Mode is w and the test is passing with wb.